apps/explorer/src/hooks/useOnScreen.ts

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import { useState, useCallback } from 'react';
+
+export const useOnScreen = ({
+    root = null,
+    rootMargin = '0px',
+    threshold = 1,
+} = {}) => {
+    const [observer, setObserver] = useState<IntersectionObserver>();
+    const [isIntersecting, setIntersecting] = useState(false);
+
+    const containerRef = useCallback(
```

Comment:
Can we share the `IntersectionObserver` object by chance. This is probably fine, however, everytime we use this hook, it will spawn 1 IO/1 observed element. If it is to track on-screen, we can just use 1 IO and track every single elements using this hook.
Also the naming is a bit deceptive since we can pass the `root` here also, meaning its more of `useIntersectionObserver` rather than a simple `useOnScreen`